enhancements/centralized-manifest-openapi-generation.md

```diff
@@ -0,0 +1,100 @@
+---
+title: centralized-manifest-openapi-generation
+authors:
+  - "@damemi"
+reviewers:
+  - "@deads2k"
+  - "@sttts"
+approvers:
+  - "@deads2k"
+  - "@sttts"
+creation-date: 2019-09-23
+last-updated: 2019-09-23
+status: provisional
+---
+
+# Centralized CRD Manifest OpenAPI Generation
+
+## Release Signoff Checklist
+- [ ] Enhancement has been determined `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is defined
+
+## Summary
+This document proposes migrating operator CRD manifests (and, more importantly, the work associated with updating them)
+to be colocated with the Golang types that define them in the [openshift/api](https://github.com/openshift/api) repository.
+
+## Motivation
+We have many operators that define custom resources, but almost all of those resources are based on Golang types defined in
+`openshift/api`. When those Golang types change, or CRD requirements change such as with [Kubernetes 1.16's
+structural schemas](https://kubernetes.io/blog/2019/06/20/crd-structural-schema/), these changes require matching updates
+to each operator repo. This is primarily evident in changes to OpenAPIV3 validation schemas, where it is not uncommon
+to make small changes to API markers which then need to be individually carried to each operator where they are generated
+into new manifest changes. In addition, maintainers currently may have different approaches to generating their manifests
+(using different tools and scripts) which can lead to discrepancies and unexpected behavior without support.
+
+By moving the operator manifests to `openshift/api`, we will colocate these types with their consumers and enable
+organization-wide changes to these manifests to be enabled at once, included with the corresponding changes to their types.
+We also will provide a single approach to maintaining the API/manifest relationship, specifically in regards to generating
+OpenAPIV3 validation schemas.
+
+### Goals
+
+- Operator and API maintainers can make changes to APIs and their corresponding CRD manifests simultaneously
+- Manifest changes can be easily read from `openshift/api` to the operators that consume them
+- Unify organization schema generation under a single tool and approach for CRD schema generation
+
+### Non-goals
+
+- We will not be responsible for maintaining individual component APIs or manifests. Though we are providing a single method
+to organize their management, maintainers will still be responsible for changes related to only their operator.
+
+## Proposal
+
+First, operator manifests will be moved to subdirectories of `openshift/api` corresponding to the names of the operators.
```

Comment:
dont' focus on operators. We don't care if they are operators.  In fact, we don't even really want the operator ones. We want the config.openshift.io ones.